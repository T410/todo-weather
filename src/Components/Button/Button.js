import React from "react";
import styles from "./Button.module.css";
import PropTypes from "prop-types";
import clsx from "clsx";
const Button = ({
	onClick,
	disabled,
	children,
	shape = "circle",
	use = "neutral",
	className,
}) => {
	const classNames = clsx(
		styles.button,
		styles[shape],
		styles[use],
		className,
		disabled ? styles.disabled : styles.enabled
	);
	return (
		<button
			data-testid="button"
			className={classNames}
			onClick={!disabled ? onClick : () => {}}
			disabled={disabled}
		>
			{children || "Button"}
		</button>
	);
};

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	disabled: PropTypes.bool,
	children: PropTypes.node.isRequired,
	shape: PropTypes.oneOf(["rounded", "circle"]),
	use: PropTypes.oneOf(["warning", "neutral", "success", "info"]),
};

export default Button;
