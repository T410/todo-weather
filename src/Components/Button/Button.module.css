@import "../../vars.css";
.button {
	display: flex;
	align-items: center;
	justify-content: center;
	border: none;
	color: white;
	text-align: center;
	transition: all 0.1s ease-in-out;
}

.button:hover {
	transform: scale(1.1);
	filter: brightness(1.1);
}

.button:active {
	transform: scale(0.9);
	filter: brightness(0.9);
}

.button * {
	display: flex !important;
}

.button:focus {
	outline: none;
}

.rounded {
	width: 6rem;
	height: 2rem;
	border-radius: 2rem;
}

.disabled {
	background-color: var(--color-disabled) !important;
}

.enabled {
	cursor: pointer;
}

.success {
	background-color: var(--color-success);
}

.warning {
	background-color: var(--color-warning);
}

.warning > svg {
	fill: #fff;
}

.neutral {
	background-color: var(--color-neutral);
}

.info {
	background-color: var(--color-info);
}

.info > svg {
	fill: #fff;
}

.circle {
	width: 2rem;
	height: 2rem;
	border-radius: 50%;
	color: var(--color-black);
}
