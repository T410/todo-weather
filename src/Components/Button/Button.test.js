import React from "react";
import { render, cleanup } from "@testing-library/react";
import Button from "./Button";
afterEach(cleanup);

it("renders correctly", () => {
	const { asFragment } = render(<Button onClick={() => {}}>Button</Button>);
	expect(
		asFragment(<Button onClick={() => {}}>Button</Button>)
	).toMatchSnapshot();
});
