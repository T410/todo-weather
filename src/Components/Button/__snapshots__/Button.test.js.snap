// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <button
    class="button circle neutral enabled"
    data-testid="button"
  >
    Button
  </button>
</DocumentFragment>
`;
